tests/test_adapter.py

```diff
@@ -17,3 +17,10 @@ def test_compatibility():
     # Make sure giving it a different version number breaks compatibility
     adapter.version = lambda: parser.version() + '1'
     assert not adapter.check_compatibility(parser)
+
+
+def test_greedy_adapter_unserializable():
+    adapter = GreedySerializeAdapter()
+    unserializable_bytes = {'key': b'\x03\xdd'}
+    s = adapter.transform(unserializable_bytes)
+    assert s == '{"key": "<<Unserializable type: bytes>>"}'
```

Comment:
We could consider base64-encoding the bytes objects, though that stretches the definition of returning data in a more usable form.